From da069396a3c85255cf09071882c220d20f4d83c8 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Fri, 21 Aug 2015 22:07:59 +0530
Subject: [PATCH] USBC-PD: Reset PD message id on power role change.

On power role change, the port partner can reset the message id.
Hence resetting the rx_msg_id on power role change.

Change-Id: I82585c455d997700b12753f488dfbef9618f95e6
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-17053
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/usb/typec/pd/protocol.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/typec/pd/protocol.c b/drivers/usb/typec/pd/protocol.c
index 869b17320602..7f397f98449c 100644
--- a/drivers/usb/typec/pd/protocol.c
+++ b/drivers/usb/typec/pd/protocol.c
@@ -49,6 +49,9 @@ static void pd_policy_update_data_role(struct pd_prot *prot,
 static void pd_policy_update_power_role(struct pd_prot *prot,
 					enum pwr_role prole)
 {
+	/* Reset the mag id on power role change*/
+	if (prot->pwr_role != prole)
+		prot->rx_msg_id = -1;
 	if (prole == POWER_ROLE_NONE || prole == POWER_ROLE_SINK)
 		prot->pwr_role = PD_POWER_ROLE_CONSUMER;
 	else if (prole == POWER_ROLE_SOURCE)
-- 
1.9.1

